site/app/controllers/admin/AutogradingConfigController.php

```diff
@@ -232,14 +233,47 @@ public function configUsedBy($config_path = null) {
     }
 
     /**
-     * @Route("/courses/{_semester}/{_course}/notebook_builder/{g_id}", methods={"GET"})
+     * @Route("/courses/{_semester}/{_course}/notebook_builder/{g_id}/{mode}", methods={"GET"})
      * @param string $g_id Gradeable ID
+     * @param string $mode The mode notebook builder should open in.  May be either 'new' or 'edit', this lets
+     * notebook builder know to save a new configuration or edit the existing one
      * @AccessControl(role="INSTRUCTOR")
      */
-    public function notebookBuilder(string $g_id) {
-        $gradeable = $this->core->getQueries()->getGradeableConfig($g_id);
+    public function notebookBuilder(string $g_id, string $mode) {
+        $gradeable = null;
+        try {
+            $gradeable = $this->core->getQueries()->getGradeableConfig($g_id);
+        }
+        catch (\Exception $exception) {
+            // Pass
+        }
+
+        // Redirect home if attempting to start notebook builder with invalid state
+        // Mostly only possible if attempting to access by manually setting invalid url parameters
+        $valid_mode = in_array($mode, ['new', 'edit']);
+        $invalid_state = $gradeable && $gradeable->isUsingDefaultConfig() && $mode === 'edit';
+        if (!$valid_mode || $invalid_state) {
+            $this->core->addErrorMessage('Invalid settings used when attempting to start Notebook Builder.');
+            return new RedirectResponse($this->core->buildUrl());
```

Comment:
Assuming the change on the check for gradeable above, and so we have a valid gradeable at this point, can redirect back to the gradeable edit page here with a more precise error message about the default configs.

Reply:
Changed